runtime/commonMain/src/kotlinx/serialization/SerialKinds.kt

```diff
@@ -26,10 +26,37 @@ import kotlinx.serialization.modules.*
  * represents a structure with two primitive fields.
  */
 public sealed class SerialKind {
+
+    /**
+     * Represents a Kotlin [Enum] with statically known values.
+     * All enum values should be enumerated in descriptor elements.
+     * Each element descriptor of a [Enum] kind represents an instance of a particular enum
+     * and has an [StructureKind.OBJECT] kind.
+     * Each [positional name][SerialDescriptor.getElementName] contains a corresponding enum element [name][Enum.name].
+     *
+     * Corresponding encoder and decoder methods are [Encoder.encodeEnum] and [Decoder.decodeEnum].
+     */
+    public object ENUM : SerialKind()
+
+    /**
+     * Represents an "unknown" type that will be known only at the moment of the serialization.
+     * Effectively it defers the choice of the serializer to a moment of the serialization, and can
+     * be used for [contextual][ContextualSerialization] serialization.
+     *
+     * To introspect descriptor of this kind, an instance of [SerialModule] is required.
+     * See [capturedKClass] extension property for more details.
+     * However, if possible options are known statically (e.g. for sealed classes), they can be
+     * enumerated in child descriptors similarly to [ENUM].
+     */
+    public object CONTEXTUAL : SerialKind()
+
     override fun toString(): String {
         // KNPE should never happen, because SerialKind is sealed and all inheritors are non-anonymous
         return this::class.simpleName!!
     }
+
+    // Provide a stable hashcode for objects
+    override fun hashCode(): Int = toString().hashCode()
```

Comment:
Why it's not stable given that all objects have identity?

Reply:
Because it uses `identityHashcode` that differs from run to run